pkg/spdk/engine.go

```diff
@@ -271,6 +285,72 @@ func (e *Engine) Create(spdkClient *spdkclient.Client, replicaAddressMap map[str
 	return e.getWithoutLock(), nil
 }
 
+// updateReplicaModeToFilterSalvageCandidates updates e.ReplicaModeMap and filters
+// out the replicas that are not eligible for salvage based on the volume head size.
+//
+// It iterates through all replicas and:
+//   - Skips the replicas with mode ERR.
+//   - Gets the volume head size of each replica.
+//   - Selects replicas with the largest volume head size as salvage candidates.
```

Comment:
Let's see if there is any better method to select candidates in the future.

cc @DamiaSan @shuo-wu